staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go

```diff
@@ -190,6 +280,19 @@ func (w *WebhookAuthorizer) Authorize(ctx context.Context, attr authorizer.Attri
 			Verb: attr.GetVerb(),
 		}
 	}
+	// Process Match Conditions before calling the webhook
+	matches, err := w.Match(ctx, r)
```

Comment:
Maybe open an issue to figure out cache semantics for pre-computed unstructured `r.Spec` so it can be reused in `eval` instead of computing it every time for `Authorize` for every webhook.

Reply:
https://github.com/kubernetes/kubernetes/issues/121578